pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -443,21 +443,15 @@ func gcListWatcher(client *dynamic.Client, resource unversioned.GroupVersionReso
 			// namespaces if it's namespace scoped, so leave
 			// APIResource.Namespaced as false is all right.
 			apiResource := unversioned.APIResource{Name: resource.Resource}
-			// The default parameter codec used by the dynamic client cannot
-			// encode api.ListOptions.
-			// TODO: api.ParameterCodec doesn't support thirdparty objects.
-			// We need a generic parameter codec.
-			return client.ParameterCodec(api.ParameterCodec).Resource(&apiResource, api.NamespaceAll).List(&options)
+			return client.ParameterCodec(dynamic.VersionedParameterEncoderWithV1Fallback).Resource(&apiResource, api.NamespaceAll).List(&options)
 		},
 		WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
 			// APIResource.Kind is not used by the dynamic client, so
 			// leave it empty. We want to list this resource in all
 			// namespaces if it's namespace scoped, so leave
 			// APIResource.Namespaced as false is all right.
 			apiResource := unversioned.APIResource{Name: resource.Resource}
-			// The default parameter codec used by the dynamic client cannot
-			// encode api.ListOptions.
-			return client.ParameterCodec(api.ParameterCodec).Resource(&apiResource, api.NamespaceAll).Watch(&options)
+			return client.ParameterCodec(dynamic.VersionedParameterEncoderWithV1Fallback).Resource(&apiResource, api.NamespaceAll).Watch(&options)
```

Comment:
Namespace controller lists and watches namespaces only, so it's *not* using the dynamic client for listing or watching: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/namespace/namespace_controller.go#L82-L88.

Same for the quota controller: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/resourcequota/resource_quota_controller.go#L93-L99